core/templates/pages/contributor-dashboard-admin-page/services/contributor-dashboard-admin-backend-api.service.ts

```diff
@@ -203,6 +204,49 @@ export class ContributorDashboardAdminBackendApiService {
         );
     });
   }
+
+  async updateQuestionRightsAsync(
+    username: string,
+    userWasGrantedQuestionSubmissionRights: boolean,
+    userWasGrantedQuestionReviewRights: boolean,
+    userCanCurrentlySubmitQuestions: boolean,
+    userCanCurrentlyReviewQuestions: boolean
+  ): Promise<void> {
+    if (
+      userWasGrantedQuestionSubmissionRights !== userCanCurrentlySubmitQuestions
+    ) {
+      if (userWasGrantedQuestionSubmissionRights) {
+        await this.addContributionReviewerAsync(
+          AppConstants.CD_USER_RIGHTS_CATEGORY_SUBMIT_QUESTION,
+          username,
+          null
+        );
+      } else {
+        await this.removeContributionReviewerAsync(
+          username,
+          AppConstants.CD_USER_RIGHTS_CATEGORY_SUBMIT_QUESTION,
```

Comment:
It seems odd that the order of args here differs from the previous branch?

Reply:
@seanlip PTAL, i wasn't sure about what was meant here, the order of the args order in updateQuestionAsync hasn't changed? Could you explain a little bit more please?

Reply:
When calling addContributionReviewerAsync, the first argument is the rights category and the second argument is the username.

When calling removeContributionReviewerAsync, the first argument is the username and the second argument is the rights category.

I was wondering, why the inconsistency; should we fix that?

Reply:
Ah I see, thanks @seanlip, not sure about the reason it was done this way, but I can go back and fix it.

Reply:
Done